### ISSUE TITLE
spring-boot-test 2.0.0M7 has strong dependency on optional Mockito

### ISSUE THREAD

**Issue by stevenschlansker** [spring-projects/spring-boot#11508]
With Spring Boot 2.0.0M7 in `MockReset`,
```java
	static MockReset get(Object mock) {
		MockReset reset = MockReset.NONE;
		MockingDetails mockingDetails = Mockito.mockingDetails(mock);
		if (mockingDetails.isMock()) {
			MockCreationSettings<?> settings = mockingDetails.getMockCreationSettings();
			List<InvocationListener> listeners = settings.getInvocationListeners();
			for (Object listener : listeners) {
				if (listener instanceof ResetInvocationListener) {
					reset = ((ResetInvocationListener) listener).getReset();
				}
			}
		}
		return reset;
	}
```

this code fails with:
```
16:52:57.741 [main] WARN  o.s.test.context.TestContextManager - Caught exception while invoking 'beforeTestMethod' callback on TestExecutionListener [org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener@6973b51b] for test method [public void com.opentable.DemoServerTest.testHelloWorld() throws java.lang.Exception] and test instance [com.opentable.DemoServerTest@64469d8]
java.lang.NoClassDefFoundError: org/mockito/Mockito
	at org.springframework.boot.test.mock.mockito.MockReset.get(MockReset.java:105)
	at org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener.resetMocks(ResetMocksTestExecutionListener.java:69)
```

This seems to be a re-hash of
https://github.com/spring-projects/spring-boot/pull/7089

Looking at that issue, the merged fix is not present on master -- seems to only be on the `1.4` release branch?

**Comment by wilkinsona** [comment]
The [commit that fixed #7089] is in 2.0 M7 so there must be something more to it.

**Comment by wilkinsona** [comment]
And the more to it is that I inadvertently [removed](https://github.com/spring-projects/spring-boot/commit/6e2dfc75f394f75d57b501cbdaf39245b6089d7b) the defensive `if`.
